src/binary/vector.rs

```diff
@@ -0,0 +1,288 @@
+use std::{
+    convert::{TryFrom, TryInto},
+    mem::size_of,
+};
+
+use serde::{Deserialize, Serialize};
+
+use super::{Binary, Error, Result};
+use crate::{spec::BinarySubtype, Bson, RawBson};
+
+const INT8: u8 = 0x03;
+const FLOAT32: u8 = 0x27;
+const PACKED_BIT: u8 = 0x10;
+
+/// A vector of numeric values. This type can be converted into a [`Binary`] of subtype
+/// [`BinarySubtype::Vector`].
+///
+/// ```rust
+/// # use bson::binary::{Binary, Vector};
+/// let vector = Vector::Int8(vec![0, 1, 2]);
+/// let binary = Binary::from(vector);
+/// ```
+///
+/// `Vector` serializes to and deserializes from a `Binary`.
```

Comment:
I don't think there's much value in treating this type as anything other than a convenience wrapper around `Binary` when it comes to serde, but we can revisit if a use case arises